tasks.py

```diff
@@ -5,79 +5,95 @@
 
 from invoke import task
 
-docs_dir = 'docs'
-build_dir = os.path.join(docs_dir, '_build')
+docs_dir = "docs"
+build_dir = os.path.join(docs_dir, "_build")
+
 
 @task
-def test(ctx, coverage=False, browse=False):
+def test(ctx, syntax=True, coverage=False, browse=False):
     import pytest
-    flake(ctx)
+
+    if syntax:
+        precommit(ctx)
+        flake(ctx)
     args = []
     if coverage:
-        args.extend(['--cov=webargs', '--cov-report=term', '--cov-report=html'])
+        args.extend(["--cov=webargs", "--cov-report=term", "--cov-report=html"])
 
     ignores = []
-    if sys.version_info < (3, ):
+    if sys.version_info < (3,):
         ignores += [
-            os.path.join('tests', 'test_aiohttpparser.py'),
-            os.path.join('tests', 'test_aiohttpparser_async_functions.py'),
+            os.path.join("tests", "test_aiohttpparser.py"),
+            os.path.join("tests", "test_aiohttpparser_async_functions.py"),
         ]
     if ignores:
         for each in ignores:
-            args.append('--ignore={0}'.format(each))
+            args.append("--ignore={0}".format(each))
     retcode = pytest.main(args)
     if coverage and browse:
-        webbrowser.open_new_tab(os.path.join('htmlcov', 'index.html'))
+        webbrowser.open_new_tab(os.path.join("htmlcov", "index.html"))
     sys.exit(retcode)
 
+
+@task
+def precommit(ctx):
+    ctx.run("pre-commit run --all-files", echo=True)
```

Comment:
`--show-diff-on-failure` is usually a good option for CI's sake (helps drive-by contributors that might not have set up the linting locally)